subtests/docker_cli/systemd/systemd.py

```diff
@@ -0,0 +1,84 @@
+"""
+Summary
+-------
+
+Operational Summary
+---------------------
+    1. Copy Unit file
+    2. Build image using predefined Dockerfile & p4321-server.py
+    3. Run image
+    4. Confirm p4321-server script is running
+    5. Stop container
+    6. cleanup
+
+"""
+
+import shutil
+import time
+from os.path import join
+from autotest.client import utils
+from autotest.client.shared.utils import is_port_free
+from dockertest.subtest import Subtest
+from dockertest.dockercmd import DockerCmd
+
+
+class systemd(Subtest):
+
+    def initialize(self):
+        super(systemd, self).initialize()
+        sysd_unitf = join(self.bindir, 'p4321.service')
+        dockerfilesrc = join(self.bindir, 'Dockerfile')
+        script_src = join(self.bindir, 'p4321-server.py')
+        self.stuff['tcp_ip'] = self.config['tcp_ip']
+        self.stuff['tcp_port'] = self.config['tcp_port']
+        self.stuff['sysd_unit_file'] = self.config['sysd_unit_file']
+        self.stuff['self.tmpdir'] = self.tmpdir
+        self.stuff['sysd_unitf_dest'] = self.config['sysd_unitf_dest']
+        self.stuff['docker_path'] = self.config['docker_path']
+        self.stuff['name'] = self.config['name']
+        shutil.copyfile(sysd_unitf, self.stuff['sysd_unitf_dest'])
+        shutil.copy(dockerfilesrc, self.stuff['self.tmpdir'])
+        shutil.copy(script_src, self.stuff['self.tmpdir'])
+        self.build_image()
+
+    def build_image(self):
+        dkrcmd = DockerCmd(self, 'build', ['--force-rm -t',
+                                           self.stuff['name'], '-f',
+                                           self.stuff['self.tmpdir']
+                                           + '/Dockerfile',
+                                           self.stuff['self.tmpdir']])
+        dkrcmd.execute()
+        time.sleep(5)
+
+    def run_once(self):
+        self.sysd_action(action='daemon-reload')
+        self.sysd_action(action='start',
+                         sysd_unit=self.stuff['sysd_unit_file'])
+        while is_port_free(self.stuff['tcp_port'], self.stuff['tcp_ip']):
+            time.sleep(2)
+        # FIXME
+        # for now, using cat but
+        # will try something similar to open() below
+        # read_port = open('/dev/tcp/localhost/4321').read().strip()
+        system_time = utils.run("cat < /dev/tcp/localhost/4321",
+                                ignore_status=False)
```

Comment:
This is a phony bash redirect token (and utils.run uses bash).  So using ``open()`` won't work,  that file doesn't exist.  The way to do it in python is using the ``socket`` module (which isn't easy).  I think we can leave this one running 'cat' for fixing later, before we merge.